src/test/regress/expected/multi_modifications.out

```diff
@@ -335,9 +380,123 @@ SELECT symbol, bidder_id FROM limit_orders WHERE id = 246;
  gm     |       247
 (1 row)
 
+-- IMMUTABLE functions are allowed -- even in returning
+UPDATE limit_orders SET symbol = UPPER(symbol) WHERE id = 246 RETURNING id, LOWER(symbol), symbol;
+ id  | lower | symbol 
+-----+-------+--------
+ 246 | gm    | GM
+(1 row)
+
 -- updates referencing non-IMMUTABLE functions are unsupported
 UPDATE limit_orders SET placed_at = now() WHERE id = 246;
 ERROR:  functions used in modification queries on distributed tables must be marked IMMUTABLE
+-- even in RETURNING
+UPDATE limit_orders SET placed_at = placed_at WHERE id = 246 RETURNING NOW();
+ERROR:  functions used in modification queries on distributed tables must be marked IMMUTABLE
 -- cursors are not supported
 UPDATE limit_orders SET symbol = 'GM' WHERE CURRENT OF cursor_name;
 ERROR:  distributed modifications must target exactly one shard
+-- check that multi-row UPDATE/DELETEs with RETURNING work
+INSERT INTO multiple_hash VALUES ('0', '1');
```

Comment:
I'd be nice to see an `UPSERT+RETURNING` test as well. But, that requires to have a unique  constraint on the partition key, which would make UPDATE tests less comprehensive with a single returning row. So, if you think it's not necessary, you can ignore this comment.